JASP-Engine/JASP/R/commonAnovaBayesian.R

```diff
@@ -0,0 +1,2299 @@
+  # Also adds a . in case the base64 is magically "Mean", "SD" or "N"
+  fixedB64 <- .v(fixed)
+  nmsB64 <- paste0(fixedB64, ".")
+  for (i in seq_along(fixed)) {
+    descriptivesTable$addColumnInfo(name = nmsB64[i], type = "string", title = fixed[i], combine = TRUE)
+  }
+
+  overTitle <- sprintf("%s%% Credible Interval", format(100 * options[["credibleInterval"]], digits = 3))
+  descriptivesTable$addColumnInfo(name = "Mean",  type = "number", format = "sf:4;dp:3")
+  descriptivesTable$addColumnInfo(name = "SD",    type = "number", format = "sf:4;dp:3")
+  descriptivesTable$addColumnInfo(name = "N",     type = "number", format = "dp:0")
+  descriptivesTable$addColumnInfo(name = "Lower", type = "number", format = "sf:4;dp:3", overtitle = overTitle)
+  descriptivesTable$addColumnInfo(name = "Upper", type = "number", format = "sf:4;dp:3", overtitle = overTitle)
+  jaspContainer[["tableDescriptives"]] <- descriptivesTable
+
+  if (errors$noVariables) 
+    return()
+
+  dependentB64 <- .v(dependent)
+
+  # order the data to show
+  dataset2 <- dataset[do.call(order, dataset[, fixedB64, drop = FALSE]), c(dependentB64, fixedB64)]
+
+  # by pasting the fixedFactors together we obtain the unique indices to group on. This excludes
+  # non-existent combinations. A "." is added to deal with the level "".
+  ind <- apply(dataset2[, fixedB64, drop = FALSE], 1L, paste0, ".", collapse = "")
+
+  # temporary function to calculate all descriptives
+  tmpFun <- function(data, fixedB64, dependentB64, cri) {
+
+    row <- list()
+    for (j in fixedB64)
+      row[[paste0(j, ".")]] <- as.character(data[1L, j])
+
+    N <- nrow(data)
+    row[["N"]] <- N
+
+    if (N == 0L) {
+
+      row[["Mean"]] <- row[["SD"]] <- NA
+
+    } else if (N == 1L) {
+
+      row[["Mean"]] <- data[[dependentB64]]
+      row[["SD"]] <- row[["Lower"]] <- row[["Upper"]] <- NA
+
+    } else {
+
+      row[["Mean"]] <- mean(data[[dependentB64]])
+      row[["SD"]]   <- stats::sd(data[[dependentB64]])
+
+      tmp <- .posteriorSummaryGroupMean(data[[dependentB64]], cri)
+      row[["Lower"]] <- tmp[["ciLower"]]
+      row[["Upper"]] <- tmp[["ciUpper"]]
+
+    }
+    return(row)
+  }
+
+  # apply tempFun on each subset defined by ind
+  rows <- by(dataset2, ind, tmpFun, fixedB64 = fixedB64,
+             dependentB64 = dependentB64, cri = options[["credibleInterval"]])
+
+  # do.call(rbind, rows) turns rows into a data.frame (from a list) for jaspResults
+  data <- do.call(rbind.data.frame, rows)
+
+  # add footnote if there are unobserved combinations
+  nObserved <- nrow(data)
+  nPossible <- prod(sapply(dataset2[, fixedB64], nlevels))
+  if (nObserved != nPossible) {
+    descriptivesTable$addFootnote(
+      symbol = "<em>Note.</em>", 
+      message = sprintf(
+        "Some combinations of factors are not observed and hence omitted (%d out of %d combinations are unobserved).",
```

Comment:
Changed to `%g` because `prod(...)` could return a double rather than an integer. I'm not sure why it's a problem that `%d` accepts logicals, aside from integers. I'd rather see NA in a string than a crash in R. It's worthwhile pointing out that `sprintf("%g", NA)` just returns `"NA"`, same goes for `paste`.

Reply:
Yeah I misread the thing I quoted. But %d complained here